src/lintsampler/lintsampler.py

```diff
@@ -0,0 +1,526 @@
+
+
+    def _setgrid(self,cells):
+        """Construct the cells for sampling.
+
+        Parameters
+        ----------
+        self : LintSampler
+            The LintSampler instance.
+        cells : tuple of arrays or list of tuples of arrays
+            If a tuple of arrays, the boundary values for a grid with dimensionality of the length
+            of the tuple. If a list of tuples of arrays, the boundary values for an arbitrary
+            number of grids with dimensionality the length of the tuples.
+
+        Returns
+        -------
+        None
+        
+        Attributes
+        -------
+        eval_type : string
+            The evaluation type, either 'gridsample' or 'freesample'
+        dim : int
+            The inferred dimensionality of the problem
+        edgearrays : list of numpy arrays
+            The arrays defining the edge of the grids
+        edgedims   : list of integers, optional, used if eval_type == 'gridsample'
+            The shape of the gridsample grid
+        ngrids : integer, optional, used if eval_type == 'freesample'
+            The number of distinct grids to be constructed
+        gridshape : list of tuple of integers, optional, used if eval_type == 'freesample'
+            The shape of each distinct grid
+        ngridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the 
+        nedgegridentries : list of integers, optional, used if eval_type == 'freesample'
+            The number of entries in the offset grids
+        x0 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the first vertex of the grids
+        x1 : list of numpy arrays, optional, used if eval_type == 'freesample'
+            The array of the last vertex of the grids
+
+        
+        """
+
+        # check the validity of the input
+        if len(cells)==0:
+            raise ValueError("LintSampler._setgrid: you must specify an evaluation domain with a tuple of arrays or a list of tuples of arrays. See documentation for details.")
+            # or, we could here drop into an adaptive grid selection
+            # or, we could sample on a unit hypercube with dimensions of the pdf
+
+        # 0. are we in the 1d case? the input is just a single tuple or 1d array.
+        # i.e. cells = np.linspace(-12,12,100)
+        if (isinstance(cells[0],float) | (isinstance(cells[0],int))): # do we need the integer option here? I think no?
+            self.eval_type='gridsample'
+
+            # override the inferred dimensionality
+            self.dim = 1
+
+            # set the arrays
+            # we are being passed the array directly
+            self.edgearrays = cells
+            self.edgedims = len(cells)
+
+
+        # 1. tuples defining the array -> make arrays, pass to gridsample
+        # i.e. cells = (np.linspace(-12,12,100),np.linspace(-4,4,50))
+        if isinstance(cells,tuple):
```

Comment:
Should we make this an `elif` so that it doesn't catch and override the case of a 1D tuple?
```suggestion
        elif isinstance(cells,tuple):
```

Reply:
My hope was that by putting them in order we would create a smooth tree, but this does feel more safe.